addon/error-consumers/api-consumer.js

```diff
@@ -0,0 +1,42 @@
+import Ember from 'ember';
+import BaseConsumer from './base-consumer';
+
+const {
+    getOwner,
+    get,
+    isEmpty,
+    computed: { alias }
+} = Ember;
+
+export default BaseConsumer.extend({
+    endpoint: alias('config.consumers.api-consumer.endpoint'),
+
+    async consume(descriptor) {
+        const ajax = getOwner(this).lookup('service:ajax');
+        const fastboot = getOwner(this).lookup('service:fastboot');
+        const isFastBoot = fastboot ? get(fastboot, 'isFastBoot') : false;
+        const req = get(this, 'fastboot.request');
+        const endpoint = get(this, 'endpoint');
+
+        if(!ajax || isEmpty(endpoint)) {
+            return false;
```

Comment:
@billdami I think we should use `assert` here to throw an error if there is no endpoint given